### PR TITLE
ROC-5573: Something an expert can still examine

### PR DESCRIPTION
**Pull request by KiranVarma02**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-5573

### Change description
Expert evidence page

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/directions-questionnaire/views/expert-evidence.njk [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{% extends "layout.njk" %}
+{% from "form.njk" import csrfProtection, radioOption, radioOptionWithHiddenControl, textArea, saveAndContinueButton, errorSummary %}
+
+{% set heading = t('Does the claim involve something an expert can still examine?') %}
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/views/expert-evidence.njk]: heading is translated in the govuktTemplate file and not needed here.

File: src/main/features/directions-questionnaire/routes/expert-evidence.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      } else {
+        const draft: Draft<DirectionsQuestionnaireDraft> = res.locals.draft
+        const user: User = res.locals.user
+
+        draft.document.expertEvidence = form.model
+
+        await new DraftService().save(draft, user.bearerToken)
+
+        if (form.model.expertEvidence.option === YesNoOption.YES.option) {
+          res.redirect(Paths.expertYesPage.evaluateUri({ externalId: res.locals.claim.externalId }))
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/routes/expert-evidence.ts]: These pages don't exist yet and may not be called this, I'd prefer just redirecting to the dashboard and letting who ever builds the next pages to do the linking.

File: src/main/features/directions-questionnaire/forms/models/expertEvidence.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { CompletableTask } from 'models/task'
+import { ValidationErrors as GlobalValidationErrors } from 'forms/validation/validationErrors'
+import { IsDefined, IsIn, IsNotEmpty, ValidateIf } from '@hmcts/class-validator'
+import { YesNoOption } from 'models/yesNoOption'
+
+export class ValidationErrors {
+  static readonly REASON_REQUIRED: string = 'Explain what is there to examine'
```
Comment (andrewwa-kainos) [inline src/main/features/directions-questionnaire/forms/models/expertEvidence.ts]: opinion: Would read better as `Explain what there is to examine`

File: src/main/features/directions-questionnaire/views/expert-evidence.njk [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{% extends "layout.njk" %}
+{% from "form.njk" import csrfProtection, radioOption, radioOptionWithHiddenControl, textArea, saveAndContinueButton, errorSummary %}
```
Comment (andrewwa-kainos) [inline src/main/features/directions-questionnaire/views/expert-evidence.njk]: You're not using `radioOptionWithHiddenControl` or `errorSummary`

File: src/test/features/directions-questionnaire/routes/expert-evidence.ts [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+import * as config from 'config'
+
+import { attachDefaultHooks } from '../../../routes/hooks'
+import { app } from '../../../../main/app'
+import { checkAuthorizationGuards } from '../../ccj/routes/checks/authorization-check'
+import { Paths } from 'directions-questionnaire/paths'
+import * as idamServiceMock from '../../../http-mocks/idam'
+import * as claimStoreServiceMock from '../../../http-mocks/claim-store'
+import { Paths as DashboardPaths } from 'dashboard/paths'
+import * as draftStoreServiceMock from '../../../http-mocks/draft-store'
```
Comment (andrewwa-kainos) [inline src/test/features/directions-questionnaire/routes/expert-evidence.ts]: Avoid relative paths please

File: src/main/features/directions-questionnaire/forms/models/expertEvidence.ts (hunk 3 above)
Comment (KiranVarma02) [reply src/main/features/directions-questionnaire/forms/models/expertEvidence.ts]: agree..updated

File: src/main/features/directions-questionnaire/routes/expert-evidence.ts (hunk 2 above)
Comment (KiranVarma02) [reply src/main/features/directions-questionnaire/routes/expert-evidence.ts]: I thought i am being more helpful :), i have written redirect test, added paths for next page to Paths. So, whoever is picking up next just have to change name to whatever they like :)....

File: src/test/features/directions-questionnaire/form/models/expertEvidence.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
+/* Allow chai assertions which don't end in a function call, e.g. expect(thing).to.be.undefined */
+/* tslint:disable:no-unused-expression */
+
+import { expect } from 'chai'
+import { Validator } from '@hmcts/class-validator'
+import { ExpertEvidence, ValidationErrors } from 'directions-questionnaire/forms/models/expertEvidence'
+import { expectValidationError } from '../../../../app/forms/models/validationUtils'
```
Comment (andrewwa-kainos) [inline src/test/features/directions-questionnaire/form/models/expertEvidence.ts]: Refactor relative imports please

File: src/test/features/directions-questionnaire/form/models/expertEvidence.ts [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  })
+
+  describe('from object', () => {
+    it('should return instance of expert evidence when passed ExpertEvidence object', () => {
+      const yes: YesNoOption = YesNoOption.YES
+      const whatToExamine: string = 'bank statements'
+
+      expect(ExpertEvidence.fromObject({ yes, whatToExamine })).to.be.instanceOf(ExpertEvidence)
+    })
+  })
```
Comment (andrewwa-kainos) [inline src/test/features/directions-questionnaire/form/models/expertEvidence.ts]: Should be some more tests for `fromObject`

File: src/test/features/directions-questionnaire/routes/expert-evidence.ts [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+          draftStoreServiceMock.resolveFind('directionsQuestionnaire')
+          draftStoreServiceMock.resolveFind('response')
+          draftStoreServiceMock.resolveSave()
+
+          await request(app)
+            .post(pagePath)
+            .set('Cookie', `${cookieName}=ABC`)
+            .send(validFormData)
+            .expect(res => expect(res).to.be.redirect.toLocation(whyExpertIsNeededPage))
+        })
```
Comment (andrewwa-kainos) [inline src/test/features/directions-questionnaire/routes/expert-evidence.ts]: Verify redirection to self witness page, too.

File: src/test/features/directions-questionnaire/form/models/expertEvidence.ts [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+      expect(errors).to.be.empty
+    })
+  })
+
+  describe('deserialize', () => {
+    it('should deserialize expert evidence to return instance of expert evidence', () => {
+      const expertEvidence: ExpertEvidence = new ExpertEvidence(YesNoOption.YES, 'bank statements')
+
+      expect(expertEvidence.deserialize(expertEvidence)).to.be.instanceOf(ExpertEvidence)
+    })
```
Comment (andrewwa-kainos) [inline src/test/features/directions-questionnaire/form/models/expertEvidence.ts]: Could use a few more deserialisation tests

**Pull request merged**